### PR TITLE
Passing article data as component props to boost perfs.

### PR DESCRIPTION
**Pull request by n1k0**
After reading http://facebook.github.io/react/docs/advanced-performance.html, I realized it makes sense to only rely on immutable data types passed as component props when using `PureRenderMixin`.

### CODE REVIEW

**Comment by n1k0** [comment]
@leplatrem review?

**Comment by Natim** [comment]
Nice :)

File: src/test/app_test.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ describe("ArticleEntry", function() {
   };
 
   beforeEach(function() {
-    view = TestUtils.renderIntoDocument(<ArticleEntry article={fakeArticle} />);
+    view = TestUtils.renderIntoDocument(<ArticleEntry {...fakeArticle} />);
```
Comment (leplatrem) [inline src/test/app_test.js:102]: Oh es6 :) works with mapping as well :)

**Comment by leplatrem** [comment]
r+

**Comment by almet** [comment]
I think I've also read that some Flux store implementation are actually relying on immutable data for everything. Maybe we should use more of these in the project?

**Comment by n1k0** [comment]
> I think I've also read that some Flux store implementation are actually relying on immutable data for everything. Maybe we should use more of these in the project?

Sure, actually I'm planning to add support for Immutable.js to DocBrown. Will investigate in the next few days.

**Pull request merged**